fortls/helper_functions.py

```diff
@@ -260,9 +260,21 @@ def find_paren_match(string: str) -> int:
 
     >>> find_paren_match('a, (b, (c, d)')
     -1
+
+    >>> find_paren_match('nt(sin(0.5))+8+len("ab((c")-3) :: y')
+    29
+
+    >>> find_paren_match("nt(sin(0.5))+8+len('ab))c')-3) :: y")
+    29
     """
     paren_count = 1
+    quote_state = {"'": False, '"': False}
     for i, char in enumerate(string):
+        if char in quote_state:
+            quote_state[char] = not quote_state[char]
+        if any(quote_state.values()):
+            continue
```

Comment:
Does this logic work with code like `len('ab"c)')`?

Reply:
I believe it doesn't.

Reply:
In LFortran we parse the string (handling the inner `"` and such correctly), and then you can locate the parentheses on top (when you see `"`, just parse the string).